### ISSUE TITLE
a bug of blastx mode in v0.7.12?

### ISSUE THREAD

**Issue by RNAer** [bbuchfink/diamond#39]
As shown in this example, search the nucleotide seq against its translation finds no hit:

```
$ cat > a.fna
>gb|U00096.2|:190-255 Escherichia coli str. K-12 substr. MG1655, complete genome
ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA
# This is the translated seq of a.fna
$ cat > a.faa
>gi|1786182|gb|AAC73112.1| thr operon leader peptide [Escherichia coli str. K-12 substr. MG1655]
MKRISTTITTTITITTGNGAG

$ ~/foo/diamond  makedb --in a.faa --db a

$ ~/foo/diamond blastx --db a --tmpdir /tmp/ --query a.fna --daa /tmp/a.daa

$ ~/foo/diamond view -a /tmp/a.daa -o test.fna.tab --outfmt sam

$ cat test.fna.tab 
@HD     VN:1.5  SO:query
@PG     PN:DIAMOND
@mm     BlastX
@CO     BlastX-like alignments
@CO     Reporting AS: bitScore, ZR: rawScore, ZE: expected, ZI: percent identity, ZL: reference length, ZF: frame, ZS: query start DNA coordinate

$ ~/foo/diamond -v
diamond v0.7.12.61
#Threads = 24
Scoring matrix = blosum62
Lambda = 0.267
K = 0.041
Gap open penalty = 11
Gap extension penalty = 1
Seg masking = 0
SSSE3 enabled.
Insufficient arguments. Use diamond -h for help.
```

**Comment by bbuchfink** [comment]
Try to turn off SEG masking (--seg no). You should also use the sensitive mode for such short sequences.

**Comment by RNAer** [comment]
Thanks. That is a short example I made. In the real case, the `--seg no` does not fix the problem:

```
$ cat test.fna
>gi|402552294:454259-455440 Mycoplasma genitalium M2288, complete genome
TTATTTTTTAATGTTTGTACTTACTCTCAAAGCATCTTCTACAATCATCAATTCCTCATTTGTACGAACT
CTAAAAATTTGATACTTGCTTTCATTTGTTGAAATTAGAGAACTATCTTGATAATTTCCAAATAAATTAC
TGTTAGTTTTAAAACCAAGTGAAGCAACTTTTTCAATTATTAATTGCACACAATAACTAGCATTTTCACC
AACTCCCCCAGTAAATACCAAGCTATCAATTTCACCTGAAAGTTGATTTAGGTATTTAGCAATATAGTCA
GCAACACGATTAACATACATTTTTATAGCAATATCATTAATTTCTGGTTTATCAAAAATATCACGCATGT
CAGAACTACCTGTTATAGCAAACATTCCACTCTTTTTATTTAATTCATTTACAACATCATTACATGAAAG
CTTTTGCTGTTCAGCAATGTAACTAACAATGGCAGGATCAATATCACCACTACGTGTTCCCATTATTAAT
CCTTCAAGGGGAGTGAATCCCATCGATGTGTTTAGTGATTTGCCTTGTTTAATCGCACAAACACTTGCAC
CATTACCAAGATGACAAACAATTAAATTAAGTGGTTTTTTATTAAACTTTTTTTCTAAAAACTCGTTAAT
GTATTTATAAGAAGTTCCATGAAAACCATATCTTCTTACTAAGTTATTTTTCTCTCAATTTTCAGGAACT
GCATAAAGATAATTTTCCCTTGGAATAGTAGTGTGAAAAGTGGTATCAAATACAGCAACATTCTTAGCAG
TTTTTATCTCTTTTAGAAAAATTTCAATAACATCTGCTTCTGGTTTATTATGAAGCGGTGCTAACTTAAT
GAATTCTTTTATTTTTGCTAGTGAATGTGTATCAACAAGAACTGCATCTGTAAAATAATTTGCTCCTTGT
ACTACTCTATGCCCTATTAGCCCAATTTCTGAAAGTTCAGTAATAATTTTGTTTTTTTTAAGCGCATTTA
AAAAATGCTTAACAGCAAGATTATGATCATTAAATTGAACCTTTTCTTCTATCTTTTTTTGATTAAATTC
AAGCTTAAAAAAACCATCAATGAAAATACGTTCACAAAGTCCTTTAGCTAGTACTTGTTTTTTATCATTA
AAAAGTTGAAATTTAATTGAACTGCTACCTGCATTAACAACCAAGATTTTGTGACTTTGCAT

>gb|CP001991.1|:624490-625683 Mycoplasma crocodyli MP145, complete genome
TTACTTTTTGCCTTCATATAATTTAATTGCATCATTTGCAATTAATAACTCTTCATTCGTTCTAATTACG
TATACTGGAATAGAACTATTTTTTGATGAAATCACTTTGTATTCATCTATTTTTGATTGGTTTAATTCAG
AATCTAATTCTAGTTTTACAATATTTATTTTATTAATTACATTTTGTCTGAAGTTAAAATCGTTTTCACC
AACGCCAGCTGTAAAAACAACAGCATCAATTTTAGGTCCTATTTTATTTAAGTAAATTGATAAATAATCA
GCTATTCTTTGGGCATATAGATCTAAAGCAAATTTTGCTCTTTCATTACCTTCTTTTGCATTTTTAGAAA
CGTCACGCATATCACTTGAAATACCAGAAACTCCTAGCAACCCGCTTTTCTTATTGAGAACATCAGTAAC
TTGTGAAATATCCATATGAGTTTGTTTAACAATAAATTCATGAATTGAAGGATCAATATCACCACTTCTT
GAACCCATCATAATTCCTGCTAAAGGTGTAAGTCCCATTGATGTATCAAATGATTTAGAATCTTTAATTG
CACAAAGACTTGCACCATTACCGATATGTAGATTTACATAAGTAACTCCGGTTTTATTAAGAATTTTTTC
AATAGTATTTGTTATATATTTATGCGAAATACCATGAGCTCCATAACGTTTAACTCCATATTCTTCAGTT
CATTGATATGGAATTGGATAAGTTGAATTAACCTTGTCTATAGTCATATGGAATGCTGTATCAAAATTAG
CAGACATTTTTGCATGAGGCATTACTTCTTTAAAGGCTTTCATAGCCATAACTGCACCAGGATTATGTAA
TGGTGCATAAATTGAAACCTTATCAATTAGTGCAATTTCTTTATCAGTTAATTTTGATGGACCAGCAAAA
TATGGACCACCTTGAACCACTCTAAAACCAATATACTCGATTTCTTTTTTGTCTTTAACTAAGTTAATTT
GGTCAAAAATTTCTAATATTTTTTCTACTGCAACTTTGTGGTTAGGCATTATTGCTTCAAATTCATATTT
TTTACCAAACTTAATAGTTATGTTTCCTTTTTCTAAAGTAATTCTTTCTGCTATACCATTAGCAATTAAA
GAAAAATCAGATTTATTAAATAAAGATAACTTAATTGATGAACTACCAGCATTAATAACTAAAATTTTTG
ACAT
$ cat db.faa 
>UniRef100_P47599 Acetate kinase n=1 Tax=Mycoplasma genitalium (strain ATCC 33530 / G-37 / NCTC 10195) RepID=ACKA_MYCGE
MQSHKILVVNAGSSSIKFQLFNDKKQVLAKGLCERIFIDGFFKLEFNQKKIEEKVQFNDH
NLAVKHFLNALKKNKIITELSEIGLIGHRVVQGANYFTDAVLVDTHSLAKIKEFIKLAPL
HNKPEADVIEIFLKEIKTAKNVAVFDTTFHTTIPRENYLYAVPENWEKNNLVRRYGFHGT
SYKYINEFLEKKFNKKPLNLIVCHLGNGASVCAIKQGKSLNTSMGFTPLEGLIMGTRSGD
IDPAIVSYIAEQQKLSCNDVVNELNKKSGMFAITGSSDMRDIFDKPEINDIAIKMYVNRV
ADYIAKYLNQLSGEIDSLVFTGGVGENASYCVQLIIEKVASLGFKTNSNLFGNYQDSSLI
STNESKYQIFRVRTNEELMIVEDALRVSTNIKK
>UniRef100_A4T166 Acetate kinase n=1 Tax=Mycobacterium gilvum (strain PYR-GCK) RepID=ACKA_MYCGI
MTRTVLVLNSGSSSVKYAVLEPDSGVLIADGIVERIGQEGGAHDHAAAMRGVFDSLAADG
HHLEDLGLVAVGHRVVHGGPDLYRPTIVDETVIDRLKELGPLAPLHNPPAVLGIEVARDA
LPDLPHVAVFDTAFFHDLPAAAATYAIDAEVARDWNIRRYGFHGTSHQYVSEQAAAFLDV
PLESLSQIVLHLGNGASVSAIVGGRPVETSMGLTPMEGLVMGTRSGDVDPGVIFYLWREA
GMPVEDIESMLNRRSGVRGLGGEIDFRELHRRIESGDDAAELAYEVYIHRLRKYIGAYLA
ILGSADVITFTAGVGENDAVVRRDALTGLAAFGIEIDEHLNDSPGRGARRISADGAPTTV
LVIPTDEELAIARACTGVLGARPGDG
>UniRef100_B2HPZ3 Acetate kinase n=1 Tax=Mycobacterium marinum (strain ATCC BAA-535 / M) RepID=ACKA_MYCMM
MSASRPNRVVLVLNSGSSSLKFQLVEPDSGMSRATGNIERIGEESSSVPDHDAALRRVFE
ILAEDDIDLQSCGLVAVGHRVVHGGKDFYEPTLLNDAVIGKLDELSPLAPLHNPPAVLCI
RVARALLPDVPHIAVFDTAFFHQLPPAAATYAIDRELADVWKIRRYGFHGTSHEYVSQQA
AEFLGKPIGDLNQIVLHLGNGASASAVAGGRPVETSMGLTPLEGLVMGTRSGDLDPGVIG
YLWRTAKLGVDEIESMLNHRSGMLGLAGERDFRRLRAMIDDGDPAAELAYDVFIHRLRKY
VGAYLAVLGHTDVVSFTAGIGEHDAAVRRDTLAGMAELGISLDERRNACPSGGARRISAD
DSPVTVLVIPTNEELAIARHCCSVLVAV
$ diamond makedb --in db.faa --db db
$ diamond blastx --db db --tmpdir /tmp/ --query test.fna --daa /tmp/a.daa --seg no  
$ diamond view -a /tmp/a.daa -o test.fna.tab
$ cat test.fna.tab  # this output is empty
```

**Comment by bbuchfink** [comment]
Sorry for the late reply. I'll fix this in the next version but it may take some time.

**Comment by bbuchfink** [comment]
This has been fixed in version 0.8.0.
